lib/src/context/context_manager.dart

```diff
@@ -0,0 +1,223 @@
+import 'package:nyxx/nyxx.dart';
+import 'package:nyxx_interactions/nyxx_interactions.dart';
+
+import '../commands.dart';
+import '../commands/chat_command.dart';
+import '../commands/message_command.dart';
+import '../commands/user_command.dart';
+import '../errors.dart';
+import '../util/view.dart';
+import 'autocomplete_context.dart';
+import 'chat_context.dart';
+import 'message_context.dart';
+import 'user_context.dart';
+
+/// Exposes methods for creating contexts from the raw event dispatched from Discord.
+///
+/// You do not need to create this class yourself; it is exposed through
+/// [CommandsPlugin.contextManager].
+class ContextManager {
+  /// The [CommandsPlugin] this [ContextManager] is attached to.
+  ///
+  /// All contexts created by this [ContextManager] will include [commands] as [IContext.commands].
+  ///
+  /// You might also be interested in:
+  /// - [IContext.commands], the property which exposes the [CommandsPlugin] to commands.
+  final CommandsPlugin commands;
+
+  /// Create a new [ContextManager] attached to a [CommandsPlugin].
+  ContextManager(this.commands);
+
+  /// Create a [MessageChatContext] from an [IMessage].
+  ///
+  /// [message] is the message that triggered the command, [contentView] is a [StringView] of the
+  /// message's content with the prefix already skipped and [prefix] is the content of the match
+  /// that was skipped.
+  ///
+  /// Throws a [CommandNotFoundException] if [message] did not match any command on [commands].
+  ///
+  /// You might also be interested in:
+  /// - [createInteractionChatContext], for creating [IChatContext]s from interaction events.
+  Future<MessageChatContext> createMessageChatContext(
+    IMessage message,
+    StringView contentView,
+    String prefix,
+  ) async {
+    ChatCommand command =
+        commands.getCommand(contentView) ?? (throw CommandNotFoundException(contentView));
```

Comment:
brackets around throws are not needed here and I dont belive they make code easier to read